cli/src/cmd/forge/coverage.rs

```diff
@@ -96,11 +104,41 @@ impl CoverageArgs {
         let project = {
             let mut project = config.ephemeral_no_artifacts_project()?;
 
-            // Disable the optimizer for more accurate source maps
-            project.solc_config.settings.optimizer.disable();
-            project.solc_config.settings.optimizer.runs = None;
-            project.solc_config.settings.optimizer.details = None;
-            project.solc_config.settings.via_ir = None;
+            if self.ir_minimum {
+                // TODO: How to detect solc version if the user does not specify a solc version in
+                // config  case1: specify local installed solc ?
+                //  case2: mutliple solc versions used and  auto_detect_solc == true
```

Comment:
I think we can ignore this this,

all options are sanitized in ethers-solc depending on version, meaning unsupported options are removed before calling solc